msm/src/witness.rs

```diff
@@ -0,0 +1,92 @@
+use ark_ff::Zero;
+use rayon::iter::{FromParallelIterator, IntoParallelIterator, ParallelIterator};
+
+/// The witness columns used by a gate of the MSM circuits.
+/// It is generic over the number of columns, N, and the type of the witness, T.
+/// It can be used to represent the different subcircuits used by the project.
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct Witness<const N: usize, T> {
```

Comment:
Nit: could you please inverse template argument order? Then it matches `[T;N]` and in practice `N` be a long constant name, so we'll have `Witness<Fp, MSM_SERIALIZATION_COLUMNS_NUM>` which is IMHO more readable than `Witness<MSM_SERIALIZATION_COLUMNS_NUM, Fp>`

Reply:
Noted for https://github.com/o1-labs/proof-systems/pull/1869